src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3067 @@
+        ++particleId;
+        if (particleId >= 8) {
+            particleId = 0;
+        }
+        fx16 sp8 = (LCRandom() % FX32_CONST(2)) - FX32_ONE;
+        fx16 sp4 = (LCRandom() % FX32_CONST(2)) + FX32_ONE;
+        fx16 sp0 = (LCRandom() % FX32_CONST(0.16015625)) - FX32_CONST(0.080078125);
+
+        for (j = 0; j < 4; ++j) {
+            SetVec(confetti->particles[i].unk_08[j], sp8 + sConfettiCoords[j].x, sp4 + sConfettiCoords[j].y, sp0 + sConfettiCoords[j].z);
+            SetVec(confetti->particles[i].unk_20, 0, 0, 0);
+            SetVec(confetti->particles[i].unk_26, (LCRandom() % FX32_CONST(0.125)) + FX32_CONST(0.125), (LCRandom() % FX32_CONST(0.125)) + FX32_CONST(0.125), (LCRandom() % FX32_CONST(0.125)) + FX32_CONST(0.125));
+            MTX_Identity44(&confetti->particles[i].unk_2C);
+        }
+
+        j = LCRandom() & 7;
+        while (j--) {
+            VEC_Fx16Add(&confetti->particles[i].unk_20, &confetti->particles[i].unk_26, &confetti->particles[i].unk_20);
+        }
+    }
+
+    SetLCRNGSeed(rand);
+    confetti->active = FALSE;
+    confetti->requestPushGxCommand = FALSE;
+    confetti->unk_8D20 = FALSE;
+    G3_MtxMode(GX_MTXMODE_POSITION_VECTOR);
+    return SysTask_CreateOnMainQueue(Task_RegisterHallOfFame_Confetti, confetti, 0);
+}
+
+static void RegisterHallOfFame_EndConfetti(SysTask *task) {
+   if (task != NULL) {
+       FreeToHeap(SysTask_GetData(task));
+       SysTask_Destroy(task);
+   }
+}
+
+static void RegisterHallOfFame_StartConfetti(SysTask *task) {
+   if (task != NULL) {
+       RegisterHofConfettiEmitterTaskData *confetti = (RegisterHofConfettiEmitterTaskData *)SysTask_GetData(task);
+       confetti->active = TRUE;
+   }
+}
+
+static void Task_RegisterHallOfFame_Confetti(SysTask *task, void *taskData) {
+   MtxFx44 sp28;
+   VecFx16 sp20;
+   RegisterHofConfettiEmitterTaskData *confetti = (RegisterHofConfettiEmitterTaskData *)taskData;
+   if (confetti->active) {
+       G3_BeginMakeDL(&confetti->gxDlInfo, confetti->gxCommand, sizeof(confetti->gxCommand));
+       G3B_PolygonAttr(&confetti->gxDlInfo, GX_LIGHTID_3, GX_POLYGONMODE_MODULATE, GX_CULL_NONE, 0x3F, 0x1F, 0);
+       G3B_MaterialColorDiffAmb(&confetti->gxDlInfo, RGB(20, 20, 20), RGB_BLACK, FALSE);
+       for (int i = 0; i < 48; ++i) {
+           confetti->particles[i].unk_08[3].y -= 85;
+           if (confetti->particles[i].unk_08[3].y <= -FX16_ONE) {
+               confetti->particles[i].unk_08[3].y = confetti->particles[i].unk_08[3].y + 2 *FX16_ONE;
+           }
+           confetti->particles[i].unk_08[0].y = confetti->particles[i].unk_08[3].y - FX16_CONST(0.1);
+           confetti->particles[i].unk_08[1].y = confetti->particles[i].unk_08[0].y;
+           confetti->particles[i].unk_08[2].y = confetti->particles[i].unk_08[3].y;
+           VEC_Fx16Add(&confetti->particles[i].unk_20, &confetti->particles[i].unk_26, &confetti->particles[i].unk_20);
+           MTX_Identity44(&confetti->particles[i].unk_2C);
```

Comment:
particles 2C is the translation matrix

Reply:
idk why i read it as "transition" matrix